### PR TITLE
Feature/Unify color variables across platform

### PR DESCRIPTION
**Pull request by yurii-lubynets**
closes https://github.com/sparkletown/internal-sparkle-issues/issues/1220

### CODE REVIEW

**Comment by whoaitsaimz** [comment]
Looking good Yurii! 

Some thoughts:

 _Icons_ and _danger_ buttons should use primary-accent--over as the text color.

$primary--live  seems like it might be useful as a generated variable. So lighten primary by 5% and make it more saturated by 5% and you get $primary--live.   I think the name should be something else but we can stick with it for now.

**Comment by goran-peoski-work** [comment]
> I think the name should be something else but we can stick with it for now.
☝️ 

@whoaitsaimz My guess is names are remnants/reflections of industry leading system that was also added to this project early on in development which we've been too slow in trying to phase it out.

It is now a decade old development "habbit" (for the lack of a better word) https://getbootstrap.com/docs/5.1/utilities/colors/

**Comment by yurii-lubynets** [comment]
> Icons and danger buttons should use primary-accent--over as the text color.
fixed.

File: src/scss/constants.scss [hunk 1]
```diff
@@ -1,11 +1,13 @@
 $primaryColor: var(--primary-color);
 $backgroundColor: var(--background-color);
 
+$primary-accent--over: #ffffff;
```
Comment (yurii-lubynets) [inline src/scss/constants.scss:4]: > Icons and danger buttons should use primary-accent--over as the text color.

based on https://github.com/sparkletown/sparkle/pull/2343#issuecomment-926239378

File: src/scss/constants.scss [hunk 2]
```diff
@@ -1,11 +1,13 @@
 $primaryColor: var(--primary-color);
```
Comment (yurii-lubynets) [inline src/scss/constants.scss:1]: the core changes are in `src/scss/constants.scss` 
all other files contain color renaming changes

File: src/scss/colors.scss [hunk 3]
```diff
@@ -2,20 +2,33 @@ $colors-enum: (
...
 
 :root {
   --primary-color-deprecated: #f6e652;
+
   --primary-content--under: #000000;
-  --primary-content--under-rgb: 0, 0, 0;
   --primary-content--over: #ffffff;
+  --primary-accent--over: #ffffff;
+
+  // TODO: add util function from hex to rgb
```
Comment (goran-peoski-work) [inline src/scss/colors.scss]: No need for comment. If function needed, most likely SCSS already provides it with https://sass-lang.com/documentation/modules/color . Normally, though, SCSS can use both intermittently, so may not even need one. 

```suggestion
```

File: src/components/atoms/ButtonNG/ButtonNG.scss [hunk 4]
```diff
@@ -9,11 +9,26 @@ $normal-button-width: 100%;
...
-  ("primary", $white, $primary--live, $primary--live-hover),
-  ("secondary", $white, opaque-white(0.2), opaque-white(0.25)),
-  ("white", $primary--live, $white, $white),
-  ("dark", $white, opaque-black(0.4), opaque-black(0.6)),
-  ("danger", $white, $input--invalid, lighten($input--invalid, 10%))
+  (
+    "primary",
+    $primary-content--over,
+    $primary-accent--under,
+    $primary--live-hover
```
Comment (goran-peoski-work) [inline src/components/atoms/ButtonNG/ButtonNG.scss:16]: Why mixed?

Is it more likely 

```scss
    $primary--content-over,
    $primary--accent-under,
    $primary--live-hover
```
or
```scss
    $primary-content--over,
    $primary-accent--under,
    $primary-live--hover
```
?

File: src/scss/global.scss [hunk 5]
```diff
@@ -1,12 +1,7 @@
...
-
-:root {
-  --primary-color: #f6e652;
-}
-
 @import "node_modules/bootswatch/dist/darkly/variables";
 @import "node_modules/bootstrap/scss/bootstrap";
 @import "node_modules/bootswatch/dist/darkly/bootswatch";
+@import "./colors.scss";
 @import "./constants.scss";
```
Comment (goran-peoski-work) [inline src/scss/global.scss:5]: All files just including `constants.scss` will not include `colors.scss`. This depends on global styles thus will show warning/error underlines in IDEs because it isn't directly included.

Suggest include `colors.scss` as first line in `constants.scss`

File: src/scss/constants.scss [hunk 6]
```diff
@@ -1,11 +1,16 @@
...
-$backgroundColor: var(--background-color);
+$primaryColor: var(--primary-color-deprecated);
+
+$primary-accent--over: var(--primary-accent--over);
+$primary-accent--under: var(--primary-accent--under);
+$primary-content--over: var(--primary-content--over);
+$primary-content-under: var(--primary-content--under);
+
+$primary--darken-20pp: var(--primary--darken-20pp);
+$primary--lighten-10pp: var(--primary--lighten-10pp);
```
Comment (goran-peoski-work) [inline src/scss/constants.scss:9]: Main colors: primary, secondary, accent ...
I don't think there will be `primary-accent` and `secondary-accent`...

Example from Material Design:
>A theme is a collection of color and typography options. Each theme includes three palettes that determine component colors:
>  - A primary palette for the color that appears most frequently throughout your application
>  - An accent, or secondary, palette used to selectively highlight key parts of your UI
>  - A warn, or error, palette used for warnings and error states

```suggestion
$accent--over: var(--primary-accent--over);
$accent--under: var(--primary-accent--under);
$content--over: var(--primary-content--over);
$content--under: var(--primary-content--under);

$primary--darken-20pp: var(--primary--darken-20pp);
$primary--lighten-10pp: var(--primary--lighten-10pp);
```

Comment (goran-peoski-work) [inline src/scss/constants.scss:9]: Adjectives, not imperative verbs to describe nouns.

```suggestion
$primary--darker-20pp: var(--primary--darken-20pp);
$primary--lighter-10pp: var(--primary--lighten-10pp);
```

**Review by goran-peoski-work — CHANGES_REQUESTED** [review]
Naming convention ambiguities and irregularities.

File: src/scss/colors.scss [hunk 7]
```diff
@@ -2,20 +2,33 @@ $colors-enum: (
...
   @each $colorName, $color in $colors-enum {
     @each $shade in $shades-enum {
       --#{$colorName}--darken-#{$shade}pp: #{darken($color, $shade * 1%)};
       --#{$colorName}--lighten-#{$shade}pp: #{lighten($color, $shade * 1%)};
     }
+
     --#{$colorName}-accent--under: #{$color};
+
+    @each $aphpa in $alphas-enum {
+      --#{$colorName}-accent--under-#{$aphpa*100}A: #{rgba(
```
[the remark this answers is not in the document]
Comment (yurii-lubynets) [reply src/scss/colors.scss]: fixed in c017a111d2208642df1a5495632869a97e366a79

File: src/scss/colors.scss [hunk 8]
```diff
@@ -2,20 +2,33 @@ $colors-enum: (
...
-  --primary-accent--over: #ffffff;
   @each $colorName, $color in $colors-enum {
     @each $shade in $shades-enum {
       --#{$colorName}--darken-#{$shade}pp: #{darken($color, $shade * 1%)};
       --#{$colorName}--lighten-#{$shade}pp: #{lighten($color, $shade * 1%)};
     }
+
     --#{$colorName}-accent--under: #{$color};
+
+    @each $aphpa in $alphas-enum {
```
[the remark this answers is not in the document]
Comment (yurii-lubynets) [reply src/scss/colors.scss]: fixed in c017a111d2208642df1a5495632869a97e366a79

File: src/scss/constants.scss (hunk 6 above)
Comment (yurii-lubynets) [reply src/scss/constants.scss]: fixed in 5981d86e2f21047b93f3b7f639362c96520be00f

Comment (yurii-lubynets) [reply src/scss/constants.scss]: fixed in be7e3130da6a0e0b2b73a6e280423b17a7df9052

File: src/components/atoms/ButtonNG/ButtonNG.scss (hunk 4 above)
Comment (yurii-lubynets) [reply src/components/atoms/ButtonNG/ButtonNG.scss]: fixed in be7e3130da6a0e0b2b73a6e280423b17a7df9052

File: src/scss/colors.scss (hunk 3 above)
Comment (yurii-lubynets) [reply src/scss/colors.scss]: Got it. I've added the util function via this module https://github.com/sparkletown/sparkle/pull/2343/files#diff-adb1222623d1ae146d1b87f9ccb2f5cc097d12e1cc51c065b8c4ccc00029f61dR1

It looks like SCSS can use both intermittently and `rgba` function needs RBG format only (can't find the way to calculate its value via var CSS)

File: src/scss/global.scss (hunk 5 above)
Comment (yurii-lubynets) [reply src/scss/global.scss:5]: Well, `colors.scss` contains some 

> :root {...} 

enhancements with color vars. It's included in `global.scss` file and that one is imported once in `src/index.tsx`. 

When I'm trying to add `colors.scss` as the first line in `constants.scss` => it's imported dozen of times
Screenshot 2021-10-04 at 13 16 48

File: src/scss/colors.scss (hunk 3 above)
Comment (goran-peoski-work) [reply src/scss/colors.scss]: URL https://github.com/sparkletown/sparkle/pull/2343/files#diff-adb1222623d1ae146d1b87f9ccb2f5cc097d12e1cc51c065b8c4ccc00029f61dR1 is entire diff of 85 files. What module?


> (can't find the way to calculate its value via var CSS)

This is the reason for SCSS existence, it pre-calculates values so that CSS doesn't have to.

File: src/scss/global.scss (hunk 5 above)
Comment (goran-peoski-work) [reply src/scss/global.scss:5]: This can only mean there could be a split between the single-include and multiple-include parts of the `colors.scss`.

Comment (goran-peoski-work) [reply src/scss/global.scss:5]: All good I guess, I now notice the split was already done in https://github.com/sparkletown/sparkle/pull/2343/files#diff-e81ecc2f161d11a623a02514686e44a4a4b3fb7a08a5c3b3fecab9895652ab95R1-R9

Comment (goran-peoski-work) [reply src/scss/global.scss:5]: Good work!

File: src/scss/colors.scss (hunk 3 above)
Comment (yurii-lubynets) [reply src/scss/colors.scss]: @goran-peoski-work and it loads the `colors.scss` file later 

Screenshot 2021-10-04 at 13 55 24

File: src/components/templates/PartyMap/components/Map/Map.scss [hunk 9]
```diff
@@ -1,8 +1,8 @@
 @import "scss/constants";
 
 $sand: #937c63;
-$white: #ffffff;
 $border-radius: 28px;
+$avatarpulse-box-shadow-color: var(--primary-accent--under-65a);
```
Comment (goran-peoski-work) [inline src/components/templates/PartyMap/components/Map/Map.scss:5]: [OOS]

A musing on the notion if there could be a constant like `$primary--under-65a` for this one.
Is it possible/viable to generate constants for `var()` strings even?

File: src/scss/colors.scss [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+$colors-enum: (
+  primary: #7c46fb,
+);
+$shades-enum: (5, 10, 14, 20, 50);
+$alphas-enum: (0.18, 0.2, 0.24, 0.35, 0.5, 0.56, 0.65, 0.8);
+
+@function hexToRGB($hex) {
+  @return red($hex), green($hex), blue($hex);
+}
```
Comment (goran-peoski-work) [inline src/scss/colors.scss:11]: Snake case convention in (S)CSS.

```suggestion
@function hex-to-rgb($hex) {
  @return red($hex), green($hex), blue($hex);
}
```

This looks like an overkill for the white and black colors. 

The idea was a function can be used / created **if needed**, but one can write these colors by hand (not even online calculator required) 🤷

File: src/scss/colors.scss [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  --primary-accent--over: #ffffff;
+
+  @each $colorName, $color in $colors-enum {
+    @each $shade in $shades-enum {
+      --#{$colorName}--darker-#{$shade}pp: #{darken($color, $shade * 1%)};
+      --#{$colorName}--lighter-#{$shade}pp: #{lighten($color, $shade * 1%)};
+    }
+
+    --#{$colorName}-accent--under: #{$color};
+    --#{$colorName}-accent--under-rbg: #{hexToRGB($color)};
```
Comment (goran-peoski-work) [inline src/scss/colors.scss:29]: Why is this needed as a prop variable on :root ? The `-rbg` version can just be a constant before `@each` since in the CSS there will already be `--#{$colorName}-accent--under` for use, no need of duplicates there.

File: src/scss/colors.scss [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+:root {
+  --primary-color-deprecated: #f6e652;
+
+  --primary-content--under: #000000;
+  --primary-content--under-rgb: #{hexToRGB(#000000)};
+  --primary-content--over: #ffffff;
+  --primary-content--over-rgb: #{hexToRGB(#ffffff)};
+  --primary-accent--over: #ffffff;
+
+  @each $colorName, $color in $colors-enum {
```
Comment (goran-peoski-work) [inline src/scss/colors.scss:22]: Snake case in (S)CSS, but here just `$name` should suffice.

```suggestion
  @each $name, $color in $colors-enum {
```

File: src/scss/constants.scss [hunk 13]
```diff
@@ -1,11 +1,16 @@
-$primaryColor: var(--primary-color);
-$backgroundColor: var(--background-color);
+$primaryColor: var(--primary-color-deprecated);
+
+$accent--over: var(--primary-accent--over);
+$accent--under: var(--primary-accent--under);
+$content--over: var(--primary-content--over);
+$content--under: var(--primary-content--under);
```
Comment (goran-peoski-work) [inline src/scss/constants.scss:6]: Palette convention applies to SCSS constants **and** CSS variables i.e. the palette convention applies to palette, no matter where used. Possibly TS/DB in the future.

```suggestion
$accent--over: var(--accent--over);
$accent--under: var(--accent--under);
$content--over: var(--content--over);
$content--under: var(--primary--under);
```

**Review by goran-peoski-work — CHANGES_REQUESTED** [review]
Excelent work. Few tidbits remain and will be ready to merge.

There is a noticeable absence of danger palette, but that can be added/adjusted in subsequent PR whenever it is ready to be added to the primary, accent and content ones.

sparkle-danger-01

File: src/components/molecules/AnnouncementMessage/AnnouncementMessage.scss [hunk 14]
```diff
@@ -86,7 +86,7 @@ $announce-box-shadow: 0px 2px 20px 5px;
     width: 20px;
     &:focus,
     &:hover {
-      color: var(--primary-color);
+      color: $primaryColor;
```
[the remark this answers is not in the document]
Comment (yurii-lubynets) [reply src/components/molecules/AnnouncementMessage/AnnouncementMessage.scss]: fixed in 8e20515ab04bf4df178d893dadb185d0e7b6126e

File: src/scss/constants.scss [hunk 15]
```diff
@@ -1,11 +1,16 @@
-$primaryColor: var(--primary-color);
-$backgroundColor: var(--background-color);
+$primaryColor: var(--primary-color-deprecated);
```
[the remark this answers is not in the document]
Comment (yurii-lubynets) [reply src/scss/constants.scss]: fixed in 8e20515ab04bf4df178d893dadb185d0e7b6126e

File: src/components/templates/PartyMap/components/Map/Map.scss (hunk 9 above)
Comment (yurii-lubynets) [reply src/components/templates/PartyMap/components/Map/Map.scss:5]: could be in a follow-up PR, not sure about generating consts for `var()`

File: src/scss/colors.scss (hunk 10 above)
Comment (yurii-lubynets) [reply src/scss/colors.scss]: fixed in 8e20515ab04bf4df178d893dadb185d0e7b6126e

Comment (yurii-lubynets) [reply src/scss/colors.scss]: removed at all, passed as RGB format

File: src/scss/colors.scss (hunk 12 above)
Comment (yurii-lubynets) [reply src/scss/colors.scss]: fixed in 8e20515ab04bf4df178d893dadb185d0e7b6126e

File: src/scss/colors.scss (hunk 11 above)
Comment (yurii-lubynets) [reply src/scss/colors.scss]: fixed in 8e20515ab04bf4df178d893dadb185d0e7b6126e

File: src/scss/constants.scss (hunk 13 above)
Comment (yurii-lubynets) [reply src/scss/constants.scss]: fixed in 64de4d5d8b89aef7293806ca968effa1a35a5aef

File: src/scss/colors.scss [hunk 16]
```diff
@@ -5,13 +5,13 @@ $shades-enum: (5, 10, 14, 20, 50);
...
-  --primary-content--over: #ffffff;
-  --primary-content--over-rgb: 0, 0, 0;
-  --primary-accent--over: #ffffff;
-  --primary-accent--under: #7c46fb;
-  --primary-accent--under-rbg: 124, 70, 251;
+  --content--under: #000000;
+  --content--under-rgb: 0, 0, 0;
+  --content--over: #ffffff;
+  --content--over-rgb: 255, 255, 255;
+  --accent--over: #ffffff;
```
Comment (goran-peoski-work) [inline src/scss/colors.scss:12]: Should there be `-rbg` named constant for this as well? For symmetry 🤷

File: src/scss/colors.scss [hunk 17]
```diff
@@ -21,7 +21,7 @@ $alphas-enum: (0.18, 0.2, 0.24, 0.35, 0.5, 0.56, 0.65, 0.8);
 
     @each $alpha in $alphas-enum {
       --#{$name}-accent--under-#{$alpha*100}a: #{rgba(
-          var(--primary-accent--under-rbg),
+          var(--accent--under-rbg),
           $alpha
         )};
```
Comment (goran-peoski-work) [inline src/scss/colors.scss]: `$colors-map` since it is in the form key: value.
No `accent` segment in the name since it comes from `$colors-map`.

```scss
 @each $name, $color in $colors-enum {
   @each $alpha in $alphas-enum {
      --#{$name}-#{$alpha*100}a: #{rgba(
          var(--#{$name}),
          $alpha
        )};
    }
}
```

File: src/scss/constants.scss [hunk 18]
```diff
@@ -1,9 +1,9 @@
...
-$accent--under: var(--primary-accent--under);
-$content--over: var(--primary-content--over);
-$content--under: var(--primary-content--under);
+$accent--over: var(--accent--over);
+$accent--under: var(--accent--under);
+$content--over: var(--content--over);
+$content--under: var(--content--under);
 
 $primary--darker-20pp: var(--primary--darker-20pp);
 $primary--lighter-10pp: var(--primary--lighter-10pp);
```
Comment (goran-peoski-work) [inline src/scss/constants.scss]:
```suggestion
```

These should be generated by

```scss
    @each $shade in $shades-enum {
      --#{$name}--darker-#{$shade}pp: #{darken($color, $shade * 1%)};
      --#{$name}--lighter-#{$shade}pp: #{lighten($color, $shade * 1%)};
    }
```

**Review by goran-peoski-work — CHANGES_REQUESTED** [review]
Final name check. There might be references in other files to double matches like `primary-accent` or `primary-content`

File: src/scss/colors.scss (hunk 17 above)
Comment (yurii-lubynets) [reply src/scss/colors.scss]: the one thing - `$color-map` contain HEX format of #7c46fb and here `rgba` requires only RBG format

File: src/scss/colors.scss (hunk 16 above)
Comment (yurii-lubynets) [reply src/scss/colors.scss:12]: added symmetry a9aad71bef3d1bd0c1cf71598d0fe6ddfede020f

File: src/scss/colors.scss (hunk 17 above)
Comment (goran-peoski-work) [reply src/scss/colors.scss]: K, this `var(--#{$name}-rgb)`  instead `var(--#{$name})`  👈  doesn't invalidate the aforementioned 2 items:

- $colors-map since it is in the form key: value.
- No accent segment in the name since it comes from $colors-map.

File: src/scss/constants.scss [hunk 19]
```diff
@@ -1,12 +1,18 @@
...
-$backgroundColor: var(--background-color);
+$primary--over: var(--primary-color-deprecated);
+
+$accent--over: var(--accent--over);
+$accent--under: var(--accent--under);
+$content--over: var(--content--over);
+$content--under: var(--content--under);
+
+$primary--darker-20pp: var(--primary--darker-20pp);
+$primary--lighter-10pp: var(--primary--lighter-10pp);
```
Comment (goran-peoski-work) [inline src/scss/constants.scss:9]: Not sure why these are added here while in 38 other places in 16 files it's just `var(--` without the need of `$constant-like-this`

Comment (yurii-lubynets) [reply src/scss/constants.scss]: removed in 0cd2600f5d7e357895b206741557c96ce99cd816

File: src/scss/colors.scss (hunk 17 above)
Comment (yurii-lubynets) [reply src/scss/colors.scss]: make sense, thank for a help
fixed in 0cd2600f5d7e357895b206741557c96ce99cd816

File: src/scss/colors.scss [hunk 20]
```diff
@@ -0,0 +1,29 @@
+$colors-enum: (
+  primary--under: #7c46fb,
+);
+$shades-enum: (5, 10, 14, 20, 50);
+$alphas-enum: (0.18, 0.2, 0.24, 0.35, 0.5, 0.56, 0.65, 0.8);
+
+:root {
+  // --(palette name)--(under/over)-(variation value)(unit name)
```
Comment (goran-peoski-work) [inline src/scss/colors.scss:8]:
```suggestion
  // --(palette name)--(under/over)-(optional description)-(variation value)(unit name)
```

**Review by goran-peoski-work — APPROVED** [review]
LGTM.

Some minor adjustments / additions to the colors can be done with a follow up PR.

✨ 🥳 🎈 🎆 ✨
